app/javascript/components/Greeting.js

```diff
@@ -0,0 +1,19 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+
+export const getGreeting = createAsyncThunk('getGreeting', async() => {
+    const API_URL = 'http://localhost:3000/api/messages';
+    const response = await fetch(API_URL);
+    const data = await response.json();
+    return [data.greeting];
+})
+
+const greetingReducer = (state = [], action) => {
+    switch(action.type) {
+        case 'getGreeting/fulfilled':
+            return action.payload;
+        default:
+            return state;
+    }
+}
```

Comment:
- [ ] This logic shouldn't be inside the `components` folder as it represents state management with Redux. Please consider moving this to the `redux` folder. 🤓 🔧 

- [ ] Also please consider moving the JSX you defined in your APP component to a separate `<Greeting>` component as per the requirements. 🤓 🔧